{
  if (failure instanceof GradleException) {
    throw (GradleException)failure;
  }
  if (failure != null) {
    throw new GradleException(action + " aborted because of an internal error.",failure);
  }
  return this;
}
